flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mongodb-cdc/src/test/java/org/apache/flink/cdc/connectors/mongodb/table/MongoDBTableFactoryTest.java

```diff
@@ -280,6 +286,99 @@ void testValidation() {
                 .hasStackTraceContaining("Unsupported options:\n\nunknown");
     }
 
+    @Test
+    public void testCopyExistingPipelineConflictWithIncrementalSnapshotMode() {
+        // test with 'initial.snapshotting.pipeline' configuration
+        try {
+            Map<String, String> properties = getAllOptions();
+            properties.put("scan.incremental.snapshot.enabled", "true");
+            properties.put(
+                    "initial.snapshotting.pipeline", "[{\"$match\": {\"closed\": \"false\"}}]");
+
+            createTableSource(SCHEMA, properties);
+        } catch (Throwable t) {
+            Assertions.assertThat(
+                            ExceptionUtils.findThrowableWithMessage(
+                                    t,
+                                    "The initial.snapshotting.*/copy.existing.* config only applies to "
+                                            + "Debezium mode, not incremental snapshot mode"))
+                    .isPresent();
```

Comment:
This check will silently pass even if no exceptions are thrown. Favor `assertThatThrownBy(...).rootCause().hasMessageContaining(...)`.